### DISCUSSION TITLE
Evaluate `IntEnum` to support bitwise flags

### DISCUSSION THREAD

**Discussion by winstxnhdw** [microsoft/pyright#9331] [Ideas]
```python
from enum import IntEnum

class Action(IntEnum):
    DANCE = 0x01
    WALK = 0x10

action = Action.DANCE | Action.WALK  # this evaluates to type of `int` when it could be evaluated to `Literal[3]`
```

Since enums are immutable, I wonder if this will ever be possible?

**Comment by erictraut** [comment]
This would require some pretty ugly special-case code in the type checker. It's technically possible, but I'm not inclined to add it unless there's stronger signal from other pyright users that this is an important use case.
